src/libraries/System.Net.NameResolution/src/System/Net/NameResolutionPal.Windows.cs

```diff
@@ -164,6 +164,15 @@ public static unsafe Task GetAddrInfoAsync(string hostName, bool justAddresses,
             {
                 state.RegisterForCancellation(cancellationToken);
             }
+            else if (errorCode == SocketError.TryAgain)
+            {
+                // WSATRY_AGAIN indicates possible problem with reachability according to docs.
+                // However, if servers are really unreachable, we would still get IOPenfing here
+                // and final result would be posted via overlapped IO.
+                // synchronous failure here may signal issue when GetAddrInfoExW does not work from
+                // impersonificated context.
+                return Task.FromException(new InvalidOperationException());
+            }
```

Comment:
Doesn't this now end up leaking resources that would have been freed by ProcessResult?

Reply:
fixed